### PR TITLE
Rearrange Stratum Timings

### PR DESCRIPTION
**Pull request by AndreaLanfranchi**
As described in #1503 there might be cases when timeouts are not triggered.

These changes include :

* Move the timing section at transport level
* Keep track of a queue of awaited responses
* Process response delays FIFO style and advance always the older
* Remove several deadline timers with their initializations and callback routines.
* Wrap all timings measurement in a single workloop_timer_elapsed routine which ticks every second

Still not perfect as it may trigger timeouts slightly late (up to 1 second additional delay) but on the other hand we get a correct sequence of response times

### CODE REVIEW

**Comment by jean-m-cyr** [comment]
A lot to digest here. Can you give a short overview of what you are trying to achieve? What issue are trying to solve? There is added complexity in this. What advantages does it bring?

**Comment by AndreaLanfranchi** [comment]
@jean-m-cyr the problem is described in #1503

**Comment by AndreaLanfranchi** [comment]
Very briefly ... if 2 or more solutions get submitted before the acceptance response of any of them drops in then we cancel timer at first acceptance (and we don't count on 2nd) which eventually will restart on a 3rd solution

Solution timing were somehow accurate as submission times were pulled from a lockfree queue on PoolManager but they are detached from timings in EthStratumClient

File: ethminer/main.cpp [hunk 1]
```diff
@@ -222,7 +222,7 @@ class MinerCLI
         app.add_option("--work-timeout", m_worktimeout,
                "Set disconnect timeout in seconds of working on the same job", true)
```
Comment (smurfy) [inline ethminer/main.cpp]: you changed that to milliseconds right?

Comment (AndreaLanfranchi) [reply ethminer/main.cpp]: Yes ... in the client is handled as ms.

Comment (smurfy) [reply ethminer/main.cpp]: So the description for the command should be changed as-well, sorry could have wrote it clearer in the first place ^^

Comment (AndreaLanfranchi) [reply ethminer/main.cpp]: Hmm ... the description and the CLI values are correct to be in seconds (who would ever want to set a work-timeout to 180 ms ?)
On the other hand, internally, all timeouts are handled in milliseconds so m_worktimeout * 1000

**Pull request merged**